Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -422,10 +392,15 @@ class TabTrayController: UIViewController {
 
         // Update the trait collection we reference in our layout delegate
         tabLayoutDelegate.traitCollection = traitCollection
+
+        // If we're displaying the menu as a modal and not a popover, make sure to dismiss here instead of 
+        // viewWillTransitionToSize to allow the animation to 'unwind' itself using the MenuPresenationAnimator.
+        displayedMenu?.dismissViewControllerAnimated(true, completion: nil)
```

Comment:
I think we need to change the tab trays implmentation of this method as we dismiss between landscape and portrait and we shouldn't on an iphone 6. We also switch to popover in landscape on the 6, when we should remain as modal.